### PR TITLE
Fix #1930, #2043: Added an alert to prompt the user to refresh Thimble.

### LINKED ISSUE #1930: Add "Updates Available" UX for new service worker cached content
In Brackets, and eventually in Thimble (#1459), we cache and serve static resources so that offline use cases work.  When a service worker has new content, we really should provide the user with some kind of UI that indicates they need to refresh the browser in order to load the new code.  It's a lot like having a "Updates Installed, Restart Required..." UX for a native app.

Depending on what the change is that's been updated in the cache, the update may or may not be critical for the code to continue to work.  An example is when we make changes to `dist/bramble.js` and Thimble needs to call some API function in Bramble that isn't available in the old version.  Running Brackets and Thimble separately means that they can drift out of compatibility until you refresh and force the cached code to get loaded.

Right now we don't do anything, and it worries me.  I'd like to add some kind of UI for the user to tell them that they need to refresh before they continue to use the app.  Once we have this, I can wire up the various bits of Brackets and Thimble to use it.

@flukeout this is probably worth doing before we update prod with all our new changes.

### PR DESCRIPTION
**Pull request by flukeout**
Should look like this if all goes well...

image

Fixes #1930

cc @humphd

### CODE REVIEW

File: public/editor/scripts/main.js [hunk 1]
```diff
@@ -38,6 +38,19 @@ require(["jquery", "bowser"], function($, bowser) {
     });
   }
 
+  Bramble.once("updatesAvailable", function() {
```
Comment (humphd) [inline public/editor/scripts/main.js]: Move all this new code down below line 65 where we check to make sure `window.Bramble` exists (it might not if we can't load `bramble.js`

File: views/editor/warnings.html [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    <h1>{{ gettext("errorUnsupportedBrowser") }}</h1>
+    <p>
+      {{ gettext("errorUnsupportedBrowserSuggestion") | instantiate | safe }}
+    </p>
+    <div title="{{ gettext("errorUnsupportedBrowserIgnoreTitle") }}" class="let-me-in">{{ gettext("errorUnsupportedBrowserIgnore") }}</div>
+  </div>
+</div>
+
+<div class="top-alert-bar hide" id="serviceworker-warning">
+  {{ gettext("refreshForUpdates") | safe }}
```
Comment (humphd) [inline views/editor/warnings.html]: I think you should do a separate string vs. trying to reuse this logic around having that event show this.  If they hit this error screen, they need to try a refresh, basically.  We should give them a button to press that says "Refresh" to download updates and try again.

**Comment by flukeout** [comment]
@humphd Added a Refresh button to the black screen of doom as well. R?

File: views/editor/bramble.html [hunk 3]
```diff
@@ -76,6 +77,36 @@
             -webkit-animation: none;
```
Comment (humphd) [inline views/editor/bramble.html]: Not part of this patch, but can you flip the order of this `animation` and `-webkit-animation` so the non-prefixed is first?

File: public/editor/scripts/main.js [hunk 4]
```diff
@@ -52,6 +52,20 @@ require(["jquery", "bowser"], function($, bowser) {
 
   Bramble.once("error", onError);
 
+  $("button.refresh-browser").on("click",function(){
+    window.location.reload();
```
Comment (humphd) [inline public/editor/scripts/main.js]: Make this do a hard-refresh by passing `true` to `reload(true);`

File: views/editor/bramble.html [hunk 5]
```diff
@@ -130,22 +161,19 @@
         <div class="thimble-spinner"></div>
         <div class="loading-message message">{{ gettext("loadingProject") }}</div>
         <div class="error-message message">
-            <p>{{ gettext("errorLoadingProject") | safe }}</p>
+            <!-- <p>{{ gettext("errorLoadingProject") | safe }}</p> -->
```
Comment (humphd) [inline views/editor/bramble.html]: What is this comment doing here?  Can we remove?

**Comment by flukeout** [comment]
@humphd - addressed your feedback. R?
